### PR TITLE
Alpine based version. Almost the same as the standard version.

### PR DESCRIPTION
**Pull request by nicolas-albert**
There is something else to do ?

### CODE REVIEW

File: library/convertigo [hunk 1]
```diff
@@ -1,11 +1,19 @@
...
 GitRepo: https://github.com/convertigo/docker
-GitCommit: c143ecb5aa888581cb128b7ac487aebf1418f3b3
+GitCommit: 1b3f1092aae7fbfd1094fb169be1e74f0902a164
 
 Tags: 7.5.0, 7.5, latest
 Architectures: amd64, arm32v7, arm64v8, i386
 Directory: 7.5/7.5.0
 
+Tags: 7.5.0-alpine, 7.5-alpine, alpine
+Architectures: amd64, arm32v7, arm64v8, i386
```
Comment (tianon) [inline library/convertigo]: Alpine doesn't actually support `arm32v7` -- you wanted `arm32v6` here.

Comment (tianon) [reply library/convertigo]: See also:
```console
$ bashbrew cat tomcat:7.0.82-jre8-alpine
Maintainers: Tianon Gravi <admwiggin@gmail.com> (@tianon), Joseph Ferguson <yosifkit@gmail.com> (@yosifkit)
Tags: 7.0.82-jre8-alpine, 7.0-jre8-alpine, 7-jre8-alpine
Architectures: amd64, arm32v6, arm64v8, i386, ppc64le, s390x
GitRepo: https://github.com/docker-library/tomcat.git
GitFetch: refs/heads/master
GitCommit: 5a78361a9ce7ef48979acbcabe2acd9342d0168a
Directory: 7/jre8-alpine
```

**Comment by nicolas-albert** [comment]
I just build this on my Raspberry pi3 (armv7) without any problem:
```
$ docker run --rm -it c8o-alpine bash
bash-4.4# apk --print arch
armhf
bash-4.4# uname -a
Linux bcdc47f49c83 4.9.66-v7+ #1059 SMP Fri Dec 1 15:38:42 GMT 2017 armv7l Linux
```
This is a bug ? 😅

**Comment by tianon** [comment]
Nope, not a bug at all -- `uname` is showing you details about your host kernel (which is `armv7l`, and which is why we never do arch detection via `uname` as a general rule).  When you do `docker run` on an image like `alpine:3.7`, it's running the first `arm` variant it finds (see https://github.com/moby/moby/issues/34875).  Lastly, `arm32v6` images are totally compatible and runnable on `arm32v7` devices (and we actually build them on an `arm64v8` device which is compatible with running `arm32` code).

**Comment by nicolas-albert** [comment]
Ah ok, I just check and it make sense:
```
$ docker pull arm32v7/tomcat:7.0.82-jre8-alpine
Error response from daemon: manifest for arm32v7/tomcat:7.0.82-jre8-alpine not found
$ docker pull arm32v6/tomcat:7.0.82-jre8-alpine
7.0.82-jre8-alpine: Pulling from arm32v6/tomcat
Digest: sha256:ccfd36623f5495b11c55dc0ca43c793d148dcb357f042afe5ff600a9d99d05bd
Status: Downloaded newer image for arm32v6/tomcat:7.0.82-jre8-alpine
```
I'll commit with arm32v6 for alpine tags.

**Comment by yosifkit** [comment]
Build test of #3872; 1d220754; `amd64` (`convertigo`):

```console

$ bashbrew build convertigo:7.5.0
Using bashbrew/cache:f28dc89286cd9ade8ac8a54f263224f2fd4a00fd468fbaccccd1278514c29025 (convertigo:7.5.0)
Tagging convertigo:7.5.0
Tagging convertigo:7.5
Tagging convertigo:latest

$ test/run.sh convertigo:7.5.0
testing convertigo:7.5.0
	'utc' [1/5]...passed
	'cve-2014--shellshock' [2/5]...passed
	'no-hard-coded-passwords' [3/5]...passed
	'override-cmd' [4/5]...passed
	'convertigo-hello-world' [5/5]....passed


$ bashbrew build convertigo:7.5.0-alpine
Using bashbrew/cache:a64b0172603a25e294cceefaa249a7a21b80cdd8153bb8163840feba955d5ca4 (convertigo:7.5.0-alpine)
Tagging convertigo:7.5.0-alpine
Tagging convertigo:7.5-alpine
Tagging convertigo:alpine

$ test/run.sh convertigo:7.5.0-alpine
testing convertigo:7.5.0-alpine
	'utc' [1/5]...passed
	'cve-2014--shellshock' [2/5]...passed
	'no-hard-coded-passwords' [3/5]...passed
	'override-cmd' [4/5]...passed
	'convertigo-hello-world' [5/5]....passed


$ bashbrew build convertigo:7.4.7
Using bashbrew/cache:5ad630356a01b2af73566dd6d39f9876657e29dc91e48ae0e3a2b1ca4caedc40 (convertigo:7.4.7)
Tagging convertigo:7.4.7
Tagging convertigo:7.4

$ test/run.sh convertigo:7.4.7
testing convertigo:7.4.7
	'utc' [1/5]...passed
	'cve-2014--shellshock' [2/5]...passed
	'no-hard-coded-passwords' [3/5]...passed
	'override-cmd' [4/5]...passed
	'convertigo-hello-world' [5/5]....passed


$ bashbrew build convertigo:7.4.7-alpine
Using bashbrew/cache:dfb01128f1c8bf4ea23aae03f4452a8200a78a5329300c22d9feb4e6ab0b08eb (convertigo:7.4.7-alpine)
Tagging convertigo:7.4.7-alpine
Tagging convertigo:7.4-alpine

$ test/run.sh convertigo:7.4.7-alpine
testing convertigo:7.4.7-alpine
	'utc' [1/5]...passed
	'cve-2014--shellshock' [2/5]...passed
	'no-hard-coded-passwords' [3/5]...passed
	'override-cmd' [4/5]...passed
	'convertigo-hello-world' [5/5]....passed

```

**Comment by nicolas-albert** [comment]
Great ! Thanks 👍 

It seems to me that it is missing the `arm32v6/convertigo` task (for the alpine arm32) ?
https://doi-janky.infosiftr.net/job/multiarch/view/images/view/convertigo/

**Comment by nicolas-albert** [comment]
The `arm32v6/convertigo` image (FROM `tomcat:alpine`) is built but the performance is very different of the `arm32v7/convertigo` image (FROM `tomcat`), on my raspberry pi 3:

Tomcat init in **13 sec** vs **2.5 sec** (x5 slower).
Convertigo webapp first init in **900 sec** vs **50 sec** (x18 slower).

It's sad that **alpine** isn't `arm32v7` (I see why here https://github.com/gliderlabs/docker-alpine/issues/298).

Do you think we should let the `arm32v6` version with this bad performance ?

**Comment by tianon** [comment]
Interesting -- I'm not sure exactly why it'd have significantly worse performance.

It's the only way you'll get an Alpine variant on `arm32`, so if that's important to you (or your users), then you don't really have much choice.  If that's not important to you, then you could remove it at your discretion.  IMO, it's better to have it available than not, but we don't really have a firm rule one way or the other.

**Comment by nicolas-albert** [comment]
I have an another question for the standard and the alpine versions:
Do I really need `gosu` and `tini` because `tomcat` image doesn't use them (just exec `catalina.sh`) ?

**Comment by tianon** [comment]
It depends.

Using `tini` is useful to reap zombies for software that doesn't handle `SIGCHLD` appropriately (often the case for Java and Ruby based software, in our experience).  Tomcat may very well need it, but we haven't seen reports of zombies in the wild there, so that's why it hasn't been added there.  See also https://github.com/docker-library/official-images#init for a bit more about `tini`.

Using `gosu` is useful when you need to do pre-initialization steps as root (such as setting permissions on a mounted volume), but then want to run the software itself as non-root for security purposes.

**Pull request merged**